### PR TITLE
1.1.3.5 Arrays: Range Queries 

### LINKED ISSUE #411: DSA 1.1.3.5 Arrays: Range Queries
## Description
Explain the construction and applications of Suffix Array.

## Task
- [ ] Documentation
- [ ] Audio
- [ ] Video

## Note
- While requesting the issue to be assigned, please mention your **NAME & BATCH NUMBER**.
- Kindly refer to [Issue Guidelines](https://github.com/girlscript/winter-of-contributing/blob/DSA/DSA/CONTRIBUTING.md#issues).
- In case you see that the issue has already been assigned, try to request for a different task within the same issue.
- Changes should be made inside the ````DSA/directory```` & ````DSA```` branch.
- Name of the file should be same as the issue.
- This issue is only for 'GWOC' contributors of 'DSA' Domain.

### PR DESCRIPTION
**Pull request by GenMech**
<hr>

## Explain the construction and applications of Suffix Array.  📜

 I have assigned for this issue and created a well documented file on it. @nimisha-yadav 
Fixes #411

<hr>

## Created a documented file of Suffix Array 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] DSA

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>
## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by GenMech** [comment]
@nimisha-yadav Have a look on this or let me know if there is any change or it's good to go.

**Comment by nimisha-yadav** [comment]
Please review @harshsri28

File: DSA/1.1 Arrays/3. Range Queries/1.1.3.5 Arrays: Range Queries.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+</div>
+<p>Following is the compressed trie. Compress Trie is obtained from standard trie by joining chains of single nodes. The nodes of a compressed trie can be stored by storing index ranges at the nodes.</p>
+<div align="center">
+    <img width="45%" alt="CompressedTrie" src="https://media.geeksforgeeks.org/wp-content/uploads/patternsearching2.png">
+</div>
+<h3>How to build a Suffix Tree for a given text?</h3>
+<p>As discussed above, Suffix Tree is compressed trie of all suffixes, so following are very abstract steps to build a suffix tree from given text.</p>
+<ol type="1">
+  <li>Generate all suffixes of given text.</li>
+  <li>Consider all suffixes as individual words and build a compressed trie.</li>
```
Comment (harshsri28) [inline DSA/1.1 Arrays/3. Range Queries/1.1.3.5 Arrays: Range Queries.md:31]: mention something about algorithm by Ukkonen for building suffix tree

**Review by harshsri28 — COMMENTED** [review]
mention about algorithm by Ukkonen for building suffix tree

File: DSA/1.1 Arrays/3. Range Queries/1.1.3.5 Arrays: Range Queries.md (hunk 1 above)
Comment (GenMech) [reply DSA/1.1 Arrays/3. Range Queries/1.1.3.5 Arrays: Range Queries.md:31]: Ok I will add it too

**Comment by GenMech** [comment]
@harshsri28 @nimisha-yadav I have added Ukkonen's Algothims as you said. Pls let me know now it is good to go

**Review by harshsri28 — APPROVED** [review]
LGTM

**Review by nimisha-yadav — CHANGES_REQUESTED** [review]
LGTM. Just change your file name to 1.1.3.5_Suffix_Array.md

**Comment by GenMech** [comment]
> LGTM. Just change your file name to 1.1.3.5_Suffix_Array.md

Okk

**Comment by GenMech** [comment]
@nimisha-yadav Done!!

**Review by nimisha-yadav — APPROVED** [review]
Congrats on getting your PR merged!

**Pull request merged**